### DISCUSSION TITLE
PostCSS custom config file or TS support

### DISCUSSION THREAD

**Discussion by mhkarimi1383** [vercel/next.js#73995] [Ideas]
### Goals

1. being able to use PostCSS just like the standard PostCSS
2. Supporting TS for everything :)

### Non-Goals

1.
2.
3.

### Background

I was setting up my project with tailwindcss but I was having issues after renaming `postcss.config.js` to `postcss.config.ts` and I was not able to use tailwind styles correctly

After reading documents about PostCSS in Next.JS https://nextjs.org/docs/pages/building-your-application/configuring/post-css I found that I have limited support for file name/type for PostCSS in Next.JS

### Proposal

Adding support for TS and other config files that PostCSS supports

    .postcssrc.js
    .postcssrc.mjs
    .postcssrc.cjs
    .postcssrc.ts
    .postcssrc.mts
    .postcssrc.cts
    postcss.config.js
    postcss.config.mjs
    postcss.config.cjs
    postcss.config.ts
    postcss.config.mts
    postcss.config.cts

Also I have tried installing `ts-node`, (Same issue https://github.com/vitejs/vite/issues/5316 And PR https://github.com/vitejs/vite/pull/7935 on Vite)

**Comment by yusifaliyevpro** [comment]
Yesterday I visited this thread, and a few hours later Next.js published v16.2, which supports .ts config.

**Comment by kachkaev** [comment]
If you use TailwindCSS v4, your postcss config will be quite small. You can avoid `postcss.*` in your root folder completely by moving the config into `package.json`:

```json
{
  "...": "...",
  "postcss": {
    "plugins": {
      "@tailwindcss/postcss": {}
    }
  }
}
```

Next.js understands that. [Example](https://github.com/kachkaev/website/blob/12d003a9bda32b94382282010e4c6b0588035474/package.json#L67-L71)
